connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -661,25 +682,35 @@ ConfigInfos validateConnectorConfig(
             }
         }
         String connType = connectorProps.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG);
+        VersionRange connVersion = null;
+        try {
+            connVersion = PluginVersionUtils.connectorVersionRequirement(connectorProps.get(CONNECTOR_VERSION));
+        } catch (Exception e) {
+            throw new BadRequestException(e.getMessage());
+        }
+
         if (connType == null)
             throw new BadRequestException("Connector config " + connectorProps + " contains no connector type");
 
-        Connector connector = getConnector(connType);
-        ClassLoader connectorLoader = plugins().connectorLoader(connType);
+        Connector connector = getConnector(connType, connVersion);
+        ClassLoader connectorLoader = connector.getClass().getClassLoader();
```

Comment:
This gets the incorrect loader, see https://github.com/apache/kafka/pull/16984#discussion_r1868461401